pkg/deviceshifu/deviceshifubase/pushtelemetry_test.go

```diff
@@ -479,3 +478,132 @@ func TestPushTelemetryCollectionService(t *testing.T) {
 		}
 	}
 }
+
+func TestInjectSecret(t *testing.T) {
+	testCases := []struct {
+		name         string
+		client       *rest.RESTClient
+		ts           *v1alpha1.TelemetryService
+		ns           string
+		specUsername string
+		specPassword string
+	}{
+		{
+			name:   "case0 no secrets found",
+			client: mockRestClientFor("", t),
+			ts: &v1alpha1.TelemetryService{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-ts",
+				},
+				Spec: v1alpha1.TelemetryServiceSpec{
+					ServiceSettings: &v1alpha1.ServiceSettings{
+						HTTPSetting: &v1alpha1.HTTPSetting{
+							Username: unitest.ToPointer("origin-username"),
+							Password: unitest.ToPointer("origin-password"),
+						},
+					},
+				},
+			},
+			ns:           "test",
+			specUsername: "origin-username",
+			specPassword: "origin-password",
+		},
+		{
+			name: "case1 have HTTP username secret",
+			client: mockRestClientFor(`{
+  "data": {
```

Comment:
format？

Reply:
I can't find a good way to format the raw string between "`" in go.